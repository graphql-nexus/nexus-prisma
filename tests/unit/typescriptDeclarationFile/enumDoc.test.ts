import dedent from 'dindist'
import { testGeneratedModules } from '../../__helpers__/testers'

testGeneratedModules({
  description:
    'When an enum has no documentation comment, then it gets the default JSDoc and its description field is null',
  databaseSchema: dedent`
    enum Foo {
      a
    }
  `,
})

testGeneratedModules({
  description:
    'When an enum has a documentation comment, then it is used for the JSDoc of that enum and its $description field',
  databaseSchema: dedent`
    /// Some documentation
    enum Foo {
      a
    }
  `,
})

testGeneratedModules({
  description:
    'When an enum has a documentation comment across lines, then they are joined and then used for the JSDoc of that field and its $description field',
  databaseSchema: `
    /// Some documentation
    /// on
    ///
    /// multiple
    /// lines
    ///
    enum SomeEnum {
      /// Some documentation
      /// on
      ///
      /// multiple
      /// lines
      ///
      a
    }
  `,
})
