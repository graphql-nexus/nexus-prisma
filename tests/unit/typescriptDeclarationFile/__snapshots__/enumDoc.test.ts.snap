// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When an enum has a documentation comment, then it is used for the JSDoc of that enum and its $description field: index.d.ts 1`] = `
"import * as Nexus from 'nexus'
import * as NexusCore from 'nexus/dist/core'

//
//
// TYPES
// TYPES
// TYPES
// TYPES
//
//

declare namespace $Types {
  // Models

  // N/A –– You have not defined any enums in your Prisma schema file.

  // Enums

  /**
    * Nexus \`enumType\` configuration based on the \`Foo\` enum in your Prisma schema.
    *
    * Some documentation
    *
    * @example
    *
    * import { enumType } from 'nexus'
    * import { Foo } from 'nexus-prisma'
    *
    * enumType(Foo)
    */
  interface Foo {
    name: 'Foo'
    description: 'Some documentation'
    members: ['a']
  }
}


//
//
// EXPORTS
// EXPORTS
// EXPORTS
// EXPORTS
//
//

//
//
// EXPORTS: PRISMA MODELS
// EXPORTS: PRISMA MODELS
// EXPORTS: PRISMA MODELS
// EXPORTS: PRISMA MODELS
//
//

// N/A –– You have not defined any enums in your Prisma schema file.

//
//
// EXPORTS: PRISMA ENUMS
// EXPORTS: PRISMA ENUMS
// EXPORTS: PRISMA ENUMS
// EXPORTS: PRISMA ENUMS
//
//

/**
  * Nexus \`enumType\` configuration based on the \`Foo\` enum in your Prisma schema.
  *
  * Some documentation
  *
  * @example
  *
  * import { enumType } from 'nexus'
  * import { Foo } from 'nexus-prisma'
  *
  * enumType(Foo)
  */
export const Foo: $Types.Foo

//
//
// EXPORTS: OTHER
// EXPORTS: OTHER
// EXPORTS: OTHER
// EXPORTS: OTHER
//
//

import { Runtime } from '../generator/runtime/settingsSingleton'

export const $settings = Runtime.changeSettings
"
`;

exports[`When an enum has no documentation comment, then it gets the default JSDoc and its description field is null: index.d.ts 1`] = `
"import * as Nexus from 'nexus'
import * as NexusCore from 'nexus/dist/core'

//
//
// TYPES
// TYPES
// TYPES
// TYPES
//
//

declare namespace $Types {
  // Models

  // N/A –– You have not defined any enums in your Prisma schema file.

  // Enums

  /**
    * Nexus \`enumType\` configuration based on the \`Foo\` enum in your Prisma schema.
    *
    * ### ️⚠️ Missing Documentation for this Enum
    * 
    * Automatically ✨ enrich this JSDoc with information about your enum
    * type by documenting it in your Prisma schema. For example:
    *
    *
    * \`\`\`prisma
    * /// Lorem ipsum dolor sit amet...
    * enum Foo {
    *   a
    * }
    * \`\`\`
    * 
    * Learn more about Prisma Schema documentation comments [here](https://www.prisma.io/docs/concepts/components/prisma-schema#comments).
    *
    * @example
    *
    * import { enumType } from 'nexus'
    * import { Foo } from 'nexus-prisma'
    *
    * enumType(Foo)
    */
  interface Foo {
    name: 'Foo'
    description: undefined
    members: ['a']
  }
}


//
//
// EXPORTS
// EXPORTS
// EXPORTS
// EXPORTS
//
//

//
//
// EXPORTS: PRISMA MODELS
// EXPORTS: PRISMA MODELS
// EXPORTS: PRISMA MODELS
// EXPORTS: PRISMA MODELS
//
//

// N/A –– You have not defined any enums in your Prisma schema file.

//
//
// EXPORTS: PRISMA ENUMS
// EXPORTS: PRISMA ENUMS
// EXPORTS: PRISMA ENUMS
// EXPORTS: PRISMA ENUMS
//
//

/**
  * Nexus \`enumType\` configuration based on the \`Foo\` enum in your Prisma schema.
  *
  * ### ️⚠️ Missing Documentation for this Enum
  * 
  * Automatically ✨ enrich this JSDoc with information about your enum
  * type by documenting it in your Prisma schema. For example:
  *
  *
  * \`\`\`prisma
  * /// Lorem ipsum dolor sit amet...
  * enum Foo {
  *   a
  * }
  * \`\`\`
  * 
  * Learn more about Prisma Schema documentation comments [here](https://www.prisma.io/docs/concepts/components/prisma-schema#comments).
  *
  * @example
  *
  * import { enumType } from 'nexus'
  * import { Foo } from 'nexus-prisma'
  *
  * enumType(Foo)
  */
export const Foo: $Types.Foo

//
//
// EXPORTS: OTHER
// EXPORTS: OTHER
// EXPORTS: OTHER
// EXPORTS: OTHER
//
//

import { Runtime } from '../generator/runtime/settingsSingleton'

export const $settings = Runtime.changeSettings
"
`;
